eng/scripts/smoketest/Get-Package-Version.ps1

```diff
@@ -0,0 +1,73 @@
+. (Join-Path $PSScriptRoot ../../common/scripts/common.ps1)
+$ErrorActionPreference = 'Continue'
+
+$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v2"
+$allPackages = Get-AllPkgProperties
+$trackTwoPackages = $allPackages.Where({ $_.IsNewSdk })
+
+$azureCorePkgInfo = $trackTwoPackages.Where({ $_.Name -eq "Azure.Core"})
+$azureCoreVer = [AzureEngSemanticVersion]::ParseVersionString($azureCorePkgInfo.Version)
+$azureCoreVer.IsPreRelease = $false
+$azureCoreVerBase = $azureCoreVer.ToString()
+
+$currentDate = Get-Date -Format "yyyyMM"
+$currentDate = [DateTime]::ParseExact($currentDate, "yyyyMM", $null)
+$azureCoreMinDate = $currentDate.addMonths(-6).ToString("yyyyMM")\
+
+$azureCorePkg = Find-Package -Name 'Azure.Core' `
+  -Source $NugetSource `
+  -AllowPrereleaseVersions `
+  -MinimumVersion "$azureCoreVerBase-alpha.$azureCoreMinDate" `
+  -MaximumVersion "$azureCoreVerBase-alphab" `
+  -Force
+
+Write-Host "Azure.Core Version: $($azureCorePkg.Version)"
+$azureCoreVerExtension = $azureCorePkg.Version.Replace($azureCoreVerBase, "")
+$azureCoreVerDateStr = $azureCoreVerExtension.split(".")[1]
```

Comment:
Just to better understand it might be worth adding a comment about what azureCoreVerExtension is expected to look like. I believe it would be something like `-alpha.20211201.01` but worth calling out.